### PR TITLE
Add `PerformanceWarning`

### PR DESCRIPTION
**Pull request by toslunar**
From the feature in CuPy (cupy/cupy#1607).

background: #5444

### CODE REVIEW

**Comment by kmaehashi** [comment]
LGTM. Found #6618 during review.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
Is `utils` appropriate? How about `chainer.warnings`?

**Comment by toslunar** [comment]
`chainer.utils` is not very appropriate, but the module has `experimental` and `nondeterministic` too.  I'd like to move them to `chainer.warnings_`.

**Comment by niboshi** [comment]
`experimental` and `nondeterministic` are functions, not warnings.
I think it's ok to leave them in `utils`.

As for `PerformanceWarning`, I'm fine with either of the following:
* Put in `chainer.warnings_` and alias as `chainer.PerformanceWarning`.
* Put in `chainer.warnings` and use as `chainer.warnings.PerformanceWarning`.

File: chainer/__init__.py [hunk 1]
```diff
@@ -20,6 +21,7 @@
 from chainer import serializers  # NOQA
 from chainer import training  # NOQA
 from chainer import variable  # NOQA
+from chainer import warnings  # NOQA
```
Comment (niboshi) [inline chainer/__init__.py]: Doesn't this overwrite `import warnings` above?

Comment (toslunar) [reply chainer/__init__.py]: Oops

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: chainer/functions/math/ndtr.py [hunk 2]
```diff
@@ -36,7 +36,8 @@ def forward_cpu(self, x):
             except ImportError:
                 warnings.warn(
                     'SciPy is not available. Forward computation of ndtr in'
-                    ' CPU can be slow without SciPy.')
+                    ' CPU can be slow without SciPy.',
+                    utils.PerformanceWarining)
```
Comment (kmaehashi) [inline chainer/functions/math/ndtr.py]: Need fix.

Comment (toslunar) [reply chainer/functions/math/ndtr.py]: Fixed.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM! Test failures are not related to this PR.

**Pull request merged**